code/game/objects/structures/watercloset.dm

```diff
@@ -47,12 +47,61 @@
 	open = !open
 	update_icon()
 
-
 /obj/structure/toilet/update_icon()
 	icon_state = "toilet[open][cistern]"
-
+	if(!anchored)
+		pixel_x = 0
+		pixel_y = 0
+		layer = OBJ_LAYER
+	else
+		if(dir == SOUTH)
+			pixel_x = 0
+			pixel_y = 8
+		if(dir == NORTH)
+			pixel_x = 0
+			pixel_y = -8
+			layer = FLY_LAYER
 
 /obj/structure/toilet/attackby(obj/item/I, mob/living/user, params)
+	if(iswrench(I))
+		var/choices = list()
+		if(cistern)
+			choices += "Stash"
+		if(anchored)
+			choices += "Disconnect"
+		else
+			choices += "Connect"
+			choices += "Rotate"
+
+		var/response = input(user, "What do you want to do?", "[src]") as null|anything in choices
+		if(!Adjacent(user) || !response)	//moved away or cancelled
+			return
+		switch(response)
+			if("Stash")
+				stash_goods(I, user)
+			if("Disconnect")
+				user.visible_message("<span class='notice'>[user] is disconnecting [src].</span>", "<span class='notice'>You are disconnecting [src]...</span>")
```

Comment:
The tenses here got a bit mixed up. `"<span class='notice'>[user] starts disconnecting [src]...</span>", "<span class='notice'>You start disconnecting [src]...</span>"`